### ISSUE TITLE
GPOS Positioning issue

### ISSUE THREAD

**Issue by Colophon-Foundry** [googlefonts/fontmake#870]
I’m having some difficulties when trying to position combining marks when in context.

When trying to compile the following in fontmake, the contextual replacement doesn’t work (there is no movement in the string at all).
```
lookup thaiContext {
	lookupflag UseMarkFilteringSet @AboveMarks;
	pos @BaseForms [maiTho-thai maiTri-thai.small]' @AscendingVowels < -250 0 0 0 >;
	pos [maiTho-thai maiTri-thai.small]' @AscendingVowels < -250 0 0 0 >;
} thaiContext;
```
I took a look at two individual builds (one from fontmake and the other from Glyphs3) and it seems that both maiTri are non-width (0) however in fontmake the zero point is on the right, whereas in Glyphs3 the zero point is on the left. This shouldn't affect it however.

Is there a way to get Glyphs3 behaviour in fontmake? Or am I setting up my lookup incorrectly?

**Comment by ohbendy** [comment]
@simoncozens I know you found a FontMake bug where the manual rules were getting written before the rest of the mark gpos rules, could it be the same problem?

**Comment by anthrotype** [comment]
what feature is that lookup being registered under?

> a FontMake bug where the manual rules were getting written before the rest of the mark gpos rules

is that this one maybe https://github.com/googlefonts/ufo2ft/issues/506 ?

If it's a matter of the relative ordering of the lookups, try adding `# Automatic Code Start` (or End) to control the order of mark and mkmk features

**Comment by ohbendy** [comment]
Yes, I think that was the thingy I remembered from Simon's adventures.

There are actually a few more rules in that lookup now, and I've moved it to the mkmk feature because it mixes rules with single-mark adjustments and rules with mkmk adjustments. And I added the # Automatic Code End flag there, which seems to work when exporting from G2 and G3. I didn't try it with FontMake, maybe Edd got that working now?
